[test-visibility] Improvements to javascript section (#22081)
DataDog/documentation: content/en/tests/setup/javascript.md

[8]: /continuous_integration/guides/rum_integration/
[9]: https://docs.cypress.io/guides/references/migration-guide#Migrating-to-Cypress-100
[10]: https://docs.cypress.io/api/plugins/after-run-api
[11]: https://docs.cypress.io/api/plugins/after-spec-api
{{% /tab %}}

{{< /tabs >}}

### How to fix "Cannot find module 'dd-trace/ci/init'" errors

> When using `dd-trace` you might encounter this error message:

Reviewer: ```suggestion
When using `dd-trace`, you might encounter the following error message:
```

```text
 Error: Cannot find module 'dd-trace/ci/init'
```

This might be because of an incorrect usage of `NODE_OPTIONS`.

For example, if your GitHub Action looks like this:
```yml
jobs:
  my-job:
    name: Run tests
    runs-on: ubuntu-latest
    env:
      NODE_OPTIONS: -r dd-trace/ci/init
    steps:
      - name: Checkout repository
        uses: actions/checkout@v3
      - name: Install node
        uses: actions/setup-node@v3
      - name: Install dependencies
        run: npm install
      - name: Run tests
        run: npm test
```

> This will **not** work because `NODE_OPTIONS` will be interpreted by every node process, including `npm install`. If you try to import `dd-trace/ci/init` before it's intalled, this step will fail.

Reviewer: ```suggestion
**Note:** This does not work because `NODE_OPTIONS` are interpreted by every node process, including `npm install`. If you try to import `dd-trace/ci/init` before it's installed, this step fails.
```

Your GitHub Action should instead look like this:
```yml
jobs:
  my-job:
    name: Run tests
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repository
        uses: actions/checkout@v3
      - name: Install node
        uses: actions/setup-node@v3
      - name: Install dependencies
        run: npm install
      - name: Run tests
        run: npm test
        env:
          NODE_OPTIONS: -r dd-trace/ci/init
```

> To sum up:
> 
> * Make sure `NODE_OPTIONS` environment variable is only set to the process running tests.
> * Specifically avoid defining `NODE_OPTIONS` in global environment variables settings in your pipeline or job definition.

Reviewer: ```suggestion
Follow these best practices:

* Make sure the `NODE_OPTIONS` environment variable is only set to the process running tests.
* Specifically avoid defining `NODE_OPTIONS` in the global environment variables settings in your pipeline or job definition.
```

#### Using Yarn 2 or later

> If you're using `yarn>=2` and a `.pnp.cjs` file you might also get the same error:

Reviewer: ```suggestion
If you're using `yarn>=2` and a `.pnp.cjs` file, you might also get the same error:
```
